knpu/get-authenticated-user.md

```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
 I want to test that this table *does* show the correct author when its set. Create
 a new scenario to describe this:
 
-     Scenario: Products show author
-       Given I am logged in as an admin
+[[[ code('4c40dd7480') ]]]
 
 Instead of just saying there are five products I'll say:
 
-     And I author 5 products 
+[[[ code('5847ac3fba') ]]]
 
 This *is* new language that will need a step definition. To save time, we can go
-directly to the products page,
+directly to the products page:
 
-     When I go to "/admin/products"
-     Then I should not see "Anonymous"
+[[[ code('6edea384d7') ]]]
 
 Since "I" - some admin user - will be the author of the products, they should all
 show "admin": none will say "Anonymous". And we will *only* have these 5 products
 because we're clearing the database between each scenario to keep things independent.
 
-Run just this new scenario by using its line number.
+Run just this new scenario by using its line number:
 
-Great - copy the `iAuthorProducts` function code and paste it into our handy `FeatureContext`
-class - near the other product function. These two functions will be similar, so
-we should reuse the logic. Copy the internals of `thereAreProducts`, make a new
-`private function createProducts`. Pass it `$count` as an argument and also an optional
-User object which will be the author for those products. Now, add an if statement
-that says, if `$author` is passed then, `$product->setAuthor()`. I already have that
-relationship setup with in Doctrine. Great!
+```bash
+$ ./vendor/bin/behat features/web/product_admin.feature:13
+```
 
-In `thereAreProducts`, change the body of this function to `$this->createProducts($count);`.
-Do the same thing in `iAuthorProducts` for now. Clearly, this is still not setting
-the author. But I want to see if it executes first and then we'll worry about setting
-the author.
+Great - copy the `iAuthorProducts()` function code and paste it into our handy `FeatureContext`
+class - near the other product function:
 
+[[[ code('a2d4b5fdea') ]]]
+
+These two functions will be similar, so we should reuse the logic. Copy the internals
+of `thereAreProducts`, make a new `private function createProducts()`. Pass it `$count`
+as an argument and also an optional User object which will be the author for those products:
+
+[[[ code('8bbd967c08') ]]]
+
+Now, add an if statement that says, if `$author` is passed then, `$product->setAuthor()`:
+
+[[[ code('954b443f9e') ]]]
+
+I already have that relationship setup with in Doctrine. Great!
+
+In `thereAreProducts()`, change the body of this function to `$this->createProducts($count);`:
+
+[[[ code('95e36de658') ]]]
+
+Do the same thing in `iAuthorProducts()` for now:
+
+[[[ code('acd4d6cbd4') ]]]
```

Comment:
And the same question, should we wrap this single line with method name like:
```php
public function iAuthorProducts($count)
{
    $this->createProducts($count);
}
```
Looks like we get it from context above